### PR TITLE
MSI Installer for CLI

### PR DESCRIPTION
**Pull request by shivamkm07**
# Description

This adds Support for MSI Installer in CLI.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1064 _

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by shivamkm07** [comment]
**NOTE:**  MSI [ProductVersion ](https://docs.microsoft.com/en-us/windows/win32/msi/productversion?redirectedfrom=MSDN) is incompatible with semver. It only supports major.minor.build format, i.e. no pre-releases or build metadata. Also it has additional constraints like maximum value of major and minor fields is 255 and build field is 65535.

So for msi installer, we can't have versions like "1.9.0-rc.1". Instead msi does allow a fourth numeric field in version, which is although ignored, but still legally allowed, like 1.9.0.1. So for rc releases, msi would be versioned the same.

In summary:
Stable Releases: REL_VERSION: 1.9.0, MSI_VERSION: 1.9.0
Pre-Releases: REL_VERSION: 1.9.0-rc.1, MSI_VERSION: 1.9.0.1

For edge builds, it would be tagged as '0.0.0.0'

**Comment by pravinpushkar** [comment]
@shivamkm07 Also update the README.md section for this ?

**Review by mukundansundar — COMMENTED** [review]
@shivamkm07 have the workflow changes been tested? If so please link it here

File: README.md [hunk 1]
```diff
@@ -57,6 +57,15 @@ Each release of Dapr CLI includes various OSes and architectures. These binary v
    * For Linux/MacOS - `/usr/local/bin`
    * For Windows, create a directory and add this to your System PATH. For example create a directory called `c:\dapr` and add this directory to your path, by editing your system environment variable.
 
+#### From the MSI Package (windows)
+
+Install windows Dapr CLI using MSI package.
+
+1. Download the [MSI Package](https://github.com/dapr/cli/releases)
+2. Run `dapr.msi` to install Dapr CLI on your system
```
Comment (pravinpushkar) [inline README.md:65]: Do we need to run this command in administrator mode or not required ?

File: .github/workflows/dapr_cli.yaml [hunk 2]
```diff
@@ -57,7 +60,7 @@ jobs:
       - name: Set up Go ${{ env.GOVER }}
         uses: actions/setup-go@v2
         with:
-          go-version: ${{ env.GOVER }}
+          go-version: ${{ env.GOVER }}
```
Comment (pravinpushkar) [inline .github/workflows/dapr_cli.yaml:63]: nit : space added at end of line

File: .github/workflows/dapr_cli.yaml [hunk 3]
```diff
@@ -35,6 +35,9 @@ jobs:
       GOARCH: ${{ matrix.target_arch }}
       GOPROXY: https://proxy.golang.org
       ARCHIVE_OUTDIR: dist/archives
+      DaprEXEDirectory: dist/windows_amd64/release
+      DaprLicenseDirectory: .github/wix
+      WIX_BIN_PATH: 'C:/Program Files (x86)/WiX Toolset v3.11/bin'
```
Comment (pravinpushkar) [inline .github/workflows/dapr_cli.yaml:40]: Wix tool set is already installed in the vm?

Comment (shivamkm07) [reply .github/workflows/dapr_cli.yaml:40]: Yes it's there by default in windows runner: https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md

**Comment by shivamkm07** [comment]
> @shivamkm07 have the workflow changes been tested? If so please link it here

Yes: https://github.com/shivamkm07/cli/actions/runs/3049019416

File: README.md (hunk 1 above)
Comment (shivamkm07) [reply README.md:65]: No administrator mode is not required.

File: .github/workflows/dapr_cli.yaml (hunk 2 above)
Comment (shivamkm07) [reply .github/workflows/dapr_cli.yaml]: Removed

**Review by mukundansundar — APPROVED** [review]
Lgtm

**Review by pravinpushkar — APPROVED** [review]
lgtm. Thanks @shivamkm07 for the clarifications.

File: .gitignore [hunk 4]
```diff
@@ -29,3 +29,8 @@ test_output.json
 
 # Go Workspaces (introduced in Go 1.18+)
 go.work
+
+#Wix files
```
Comment (shubham1172) [inline .gitignore:33]:
```suggestion
# Wix files
```

File: .github/wix/license.rtf [hunk 5]
```diff
@@ -0,0 +1,416 @@
+{\rtf1\adeflang1025\ansi\ansicpg1252\uc1\adeff31507\deff0\stshfdbch31506\stshfloch31506\stshfhich31506\stshfbi31507\deflang16393\deflangfe16393\themelang16393\themelangfe0\themelangcs0{\fonttbl{\f2\fbidi \fmodern\fcharset0\fprq1{\*\panose 02070309020205020404}Courier New;}{\f34\fbidi \froman\fcharset0\fprq2{\*\panose 02040503050406030204}Cambria Math;}
```
Comment (shubham1172) [inline .github/wix/license.rtf:1]: Curious to know, how did you generate this RTF file? Is the `themedata` at the bottom required?

**Review by shubham1172 — APPROVED** [review]
